.github/workflows/lintdiff.yaml

```diff
@@ -0,0 +1,68 @@
+name: LintDiff (preview)
+
+on: pull_request
+
+permissions:
+  contents: read
+
+jobs:
+  lintdiff:
+    name: LintDiff (preview)
+    runs-on: ubuntu-24.04
+
+    steps:
+      - name: Checkout eng
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            eng/
+            .github/
+
+      - name: Checkout 'after' state
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+          path: after
+
+      - name: Checkout 'before' state
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.base.sha }}
+          path: before
+
+      - name: Setup Node and install deps
```

Comment:
Add `eng/tools/lint-diff` to `eng/tools/package.json`, then it will be included transitively in the root `package.json`, so users (and CI) only needs to run `npm ci` at the repo root to install all tools.